addons/mail/static/src/new/discuss/channel_member_list.js

```diff
@@ -0,0 +1,55 @@
+/** @odoo-module **/
+
+import { Component, onWillUpdateProps, onWillStart } from "@odoo/owl";
+import { useMessaging } from "@mail/new/messaging_hook";
+import { PartnerImStatus } from "./partner_im_status";
+import { Partner } from "../core/partner_model";
+
+export class ChannelMemberList extends Component {
+    setup() {
+        this.messaging = useMessaging();
+        onWillStart(() => {
+            this.getChannelMembers(this.props);
+        });
+        onWillUpdateProps((nextProps) => this.getChannelMembers(nextProps));
+    }
+    async getChannelMembers(props) {
+        const channel = props.thread.serverData.channel;
+        if (channel.channel_type === "channel") {
```

Comment:
Not sure why limit to `channel.channel_type === "channel"` here. This component makes sense to any `mail.channel`, even if only use in some channels.